File: 1.BinaryToDecimal/CPP/src/BinaryToDecimalTest.cpp
#include "../include/BinaryToDecimal.hpp"
#include "../include/BinaryToDecimalTest.hpp"

bool BinaryToDecimalTester::test(const std::vector<std::pair<std::string, int>>& testCases) {
    BinaryToDecimal btd;
    
    for (const auto& testCase : testCases) {
        const std::string& binary = testCase.first;
        int expectedDecimal = testCase.second;
        
        int result = btd.binaryToDecimal(binary);
        
        if (result != expectedDecimal) {
            return false;
        }
    }
    
    return true;
}

File: 1.BinaryToDecimal/PYTHON/BinaryToDecimalTest.py
from BinaryToDecimal import BinaryToDecimal

class BinaryToDecimalTest:
    def __init__(self):
        self.converter = BinaryToDecimal()
        self.util = Util()
        self.config = self.util.read_json('conf/config.json')
        self.testcases = self.util.read_json(self.config.get('testcasesFile'))

    def test(self, input, expected_output):
        # Convert the input to decimal using the BinaryToDecimal class
        result = self.converter.binary_to_decimal(input)

        # Compare the result to the expected output
        return result == expected_output
